app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/provision_workflow.rb

```diff
@@ -22,8 +22,53 @@ def volume_dialog_keys
     %i[name size shareable]
   end
 
+  def vm_image
+    template_id = values&.dig(:src_vm_id, 0)
+    ar_ems.miq_templates.select { |t| t.id == template_id }[0]
+  end
+
+  def sap_image?
+    vm_image.description == 'stock-sap'
+  end
+
+  def sap_flavor
+    if sap_image?
+      selected_flavor_name = values&.dig(:sys_type, 1)
+      ar_ems.flavors.select { |flavor| flavor.name == selected_flavor_name }[0]
+    end
+  end
+
+  def sap_flavor_memory(_options = {})
+    if sap_flavor
+      {"2" => sap_flavor.memory}
+    end
+  end
+
+  def sap_flavor_cpus(_options = {})
+    if sap_flavor
+      {sap_flavor.cpus.to_s => ''}
+    end
+  end
+
+  def allowed_instance_type(_options = {})
+    if sap_image?
+      {
+        0 => "dedicated"
+      }
+    else
+      {
+        0 => "capped",
+        1 => "shared",
+        2 => "dedicated"
+      }
+    end
+  end
+
   def allowed_sys_type(_options = {})
-    ar_sys_types = ar_ems.flavors
+    flavor_type = sap_image? ? "ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::SAPProfile" : "ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::SystemType"
```

Comment:
Super minor but if you want to save some horizontal space you can do
`"ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::#{sap_image? ? "SAPProfile" : "SystemType"}"`